src/transmog_scripts.cpp

```diff
@@ -1201,7 +1201,17 @@ class WS_Transmogrification : public WorldScript
         {
             LOG_INFO("module", "Loading transmog appearance collection cache....");
             uint32 collectedAppearanceCount = 0;
-            QueryResult result = CharacterDatabase.Query("SELECT account_id, item_template_id FROM custom_unlocked_appearances");
+
+
+            std::string query = "SELECT account_id, item_template_id FROM custom_unlocked_appearances";
+
+            if (sConfigMgr->GetOption<bool>("Transmogrification.EnableSortByQualityAndName", true)) {
+                const std::string acore_world_name = sConfigMgr->GetOption<std::string>("Transmogrification.AcoreWorldName", "acore_world");
+                query = "SELECT custom_unlocked_appearances.account_id, custom_unlocked_appearances.item_template_id, " + acore_world_name + ".item_template.name, " + acore_world_name + ".item_template.Quality FROM custom_unlocked_appearances INNER JOIN " + acore_world_name + ".item_template ON custom_unlocked_appearances.item_template_id=" + acore_world_name + ".item_template.entry ORDER BY " + acore_world_name + ".item_template.Quality DESC, " + acore_world_name + ".item_template.name ASC;";
```

Comment:
This query will perform very badly with this cross-db join

Reply:
I know, I do not know how to improve it
would be better to convert it into a left join?